src/main/java/org/broadinstitute/hellbender/utils/hdf5/HDF5File.java

```diff
@@ -1,1052 +0,0 @@
-package org.broadinstitute.hellbender.utils.hdf5;
```

Comment:
Yes, it should all be in https://github.com/broadinstitute/hdf5-java-bindings

In my opinon, there are a few methods that got stranded in this repo (like some things you pointed out, like checking that the library is loaded or whether a file can be read as HDF5), as well as a few PoN-specific things that made their way into that repo that could be removed.  But we can sort this out later.